app/src/views/AnalysisManagementView.js

```diff
@@ -181,6 +182,24 @@
             this.editor.setFontSize(14);
             this.editor.renderer.$cursorLayer.element.style.opacity = 0;
 
+            // 'input' instead of 'change' because it's behind a timeout so
+            // it gets fired less frequently.
+            this.editor.on('input', _.bind(function () {
+                if (this.editor.savedVersion !== this.editor.getValue()) {
+                    $("#save").addClass("btn-primary");
+                    $("#save").removeClass("disabled");
+                    if (window.onbeforeunload === null) {
+                        window.onbeforeunload = function (e) {
+                            return "You have unsaved changes in the editor.";
+                        };
+                    }
+                } else {
+                    window.onbeforeunload = null;
+                    $("#save").removeClass("btn-primary");
+                    $("#save").addClass("disabled");
```

Comment:
Not a big deal, but just for future reference, jquery provides nice chainable semantics to reduce the need to run the same selector multiple times in a row, so the above two lines could be combined into

    $('#save').removeClass("btn-primary").addClass("disabled");